scripts/tables.sql

```diff
@@ -34,3 +34,29 @@ CREATE TABLE IF NOT EXISTS user_daily_visits (
 
 CREATE UNIQUE INDEX IF NOT EXISTS unique_user_daily_idx ON user_daily_visits (user_id, device_id, visit_ts);
 CREATE INDEX IF NOT EXISTS user_daily_visit_ts_idx ON user_daily_visits (visit_ts);
+
+/** User Monthly Visits **/
+CREATE MATERIALIZED VIEW IF NOT EXISTS user_monthly_visits AS
+SELECT device_id,
+date_trunc('month', visit_ts) as month,
+user_id,
+user_agent,
+instance,
+domain,
+CASE
+WHEN user_agent LIKE '%Mozilla%Mac OS%Safari%' THEN 'Safari Mac OS'
+WHEN user_agent LIKE '%Mozilla%Windows%Firefox%' THEN 'Firefox Windows'
+WHEN user_agent LIKE '%Mozilla%Windows%Chrome%' THEN 'Chrome Windows'
+WHEN user_agent LIKE '%Mozilla%Mac OS%Firefox%' THEN 'Firefox Mac OS'
+WHEN user_agent LIKE '%Mozilla%Mac OS%Chrome%' THEN 'Chrome Mac OS'
+WHEN user_agent LIKE '%Mozilla%Linux%Firefox%' THEN 'Firefox Linux'
+WHEN user_agent LIKE '%Mozilla%Linux%Chrome%' THEN 'Chrome Linux'
+WHEN user_agent LIKE '%Tchap%NEO%' THEN 'Tchap Android NEO'
+WHEN user_agent LIKE '%Tchap%Android%' THEN 'Tchap Android'
+WHEN user_agent LIKE '%RiotNSE%iOS%' THEN 'Tchap iOS'
+WHEN user_agent LIKE '%Tchap%iOS%' THEN 'Tchap iOS'
+WHEN user_agent LIKE '%Element%' THEN 'Element'
+ELSE 'Autre'
+END as device_type
+FROM user_daily_visits
+GROUP BY device_id, month, user_id, user_agent, instance, domain, device_type;
```

Comment:
```suggestion
CREATE MATERIALIZED VIEW IF NOT EXISTS user_monthly_visits AS
SELECT device_id,
date_trunc('month', visit_ts) as month,
user_id,
instance,
domain,
CASE
WHEN user_agent LIKE '%Mozilla%Mac OS%Safari%' THEN 'Safari Mac OS'
WHEN user_agent LIKE '%Mozilla%Windows%Firefox%' THEN 'Firefox Windows'
WHEN user_agent LIKE '%Mozilla%Windows%Chrome%' THEN 'Chrome Windows'
WHEN user_agent LIKE '%Mozilla%Mac OS%Firefox%' THEN 'Firefox Mac OS'
WHEN user_agent LIKE '%Mozilla%Mac OS%Chrome%' THEN 'Chrome Mac OS'
WHEN user_agent LIKE '%Mozilla%Linux%Firefox%' THEN 'Firefox Linux'
WHEN user_agent LIKE '%Mozilla%Linux%Chrome%' THEN 'Chrome Linux'
WHEN user_agent LIKE '%Tchap%NEO%' THEN 'Tchap Android NEO'
WHEN user_agent LIKE '%Tchap%Android%' THEN 'Tchap Android'
WHEN user_agent LIKE '%RiotNSE%iOS%' THEN 'Tchap iOS'
WHEN user_agent LIKE '%Tchap%iOS%' THEN 'Tchap iOS'
WHEN user_agent LIKE '%Element%' THEN 'Element'
ELSE 'Autre'
END as device_type
FROM user_daily_visits
GROUP BY device_id, month, user_id, instance, domain, device_type;
```
Proposition to solve duplicate in case the user agent change during the month